### ISSUE TITLE
PreValueEditor contenttypepicker.hml no longer works in 8.5.5.

### ISSUE THREAD

**Issue by John-Blair** [umbraco/Umbraco-CMS#8059]
A brief description of the issue goes here.

In my property editor I use the prevalues "contentypepicker" view , but this no longer works in 8.5.5 - it used to work in an earlier version of 8.

## Umbraco version

I am seeing this issue on Umbraco version: 8.5.5

Reproduction
------------

The property editor is defined in the nuget package JB.Taxonomy.

From the package manifest:
` "prevalues": {
                "fields":[
                    {
                        "label":"Tag Root Folder",
                        "description": "The folder containing all the tags. They can be in a nested folder structure.",
                        "key":  "startNodeId",
                        "view": "treepicker"
                    },
                    {
                        "label":"Tag Type",
                        "description": "The content type used for selecting tags and creating new ones.",
                        "key":"tagTypeAlias",
                        "view": "contenttypepicker"
                    }
                ]
            }`

If you're filing a bug, please describe how to reproduce it. Include as much
relevant information as possible, such as:

### Bug summary

The contenttypepicker used to provide a popup of content types to select from like in the following picture - but clicking Add now redirects me to the home content node - no popup.

![image](https://user-images.githubusercontent.com/28809458/80965545-98a52b80-8e0a-11ea-8344-3c888a0a96e4.png)

### Specifics

No error message displayed and nothing appears in the console in Chrome to indicate an error.

In 8.5.5. umbraco the current value of contenttype is no longer displayed and Add does not display a popup to select the content type.

### Steps to reproduce

Create a property editors and use the contenttyppicker prevalue - from the file umbraco\views\prevalueeditors\contenttypepicker.html

### Expected result

Popup like screenshot above.

### Actual result

Described above.

**Comment by John-Blair** [comment]
I've just done a clean install of umbraco 8.6.1 and I notice the prevalue editor "contenttypepicker.html" is no longer part of the installation.

Is there an upgrade path for this breaking change?

**Comment by nul800sebastiaan** [comment]
I don't think we've explicitly documented that this would be a breaking change, only changing the location of CSS or image files are mentioned: https://our.umbraco.com/Documentation/Development-Guidelines/breaking-changes#ui-files

One could argue that `Changes to HTML/Markup are not considered breaking changes` means we are allowed to remove html files, but it's a grey area. I wonder what you were using this for?

In any case, the file was moved to treesourcetypepicker.html in this PR:  https://github.com/umbraco/Umbraco-CMS/pull/6177/files#diff-7dccf7eda27eca42feea35ca15501718 

So hopefully you can use that one going forward.

**Comment by John-Blair** [comment]
@nul800sebastiaan ,
The package manifest "view" property requires you to specify the **name** of a file. If you rename or delete that file then it is clearly a breaking change - there is no grey.

I have a workaround but my nuget package users all had broken installations.

Note: The  "treesourcetypepicker" does not work - clicking on the Add button does nothing - you are better off removing that file given it doesn't work.

My Taxonomy property editor was allowing my users to choose the name of a Document Type to be created for each new tag e.g. Currency, Investor Type, Fund Manager etc.

I'm happy for this to remain closed. Thanks.

**Comment by nul800sebastiaan** [comment]
Thanks @John-Blair. I totally understand that for YOU it's entirely clear, it's breaking. But we have set up some rules in that doc I linked to for what is technically a breaking change, but we don't consider it as such. So for me it is a grey area, since I don't know if we should explicitly mark that as a breaking change, I will try to get some clarification.
I understand that this can be frustrating, but this is a trade-off between us being able to have some flexibility or us having to release a new major version every 6 weeks. We try to make it predictable what areas CAN break so you don't explicitly rely on those areas.

The `treesourcetypepicker` is actively being used so I don't think we should remove it though! 😉
